### PR TITLE
Replace POST http 'identifiables-count' method by a GET

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 1]
```diff
@@ -695,10 +696,10 @@ public Optional<List<IdentifiableAttributes>> exportFilter(UUID id, UUID network
...
 
-    public Map<String, List<Long>> getIdentifiablesCount(Map<String, List<UUID>> ids, UUID networkUuid, String variantId) {
-        Objects.requireNonNull(ids);
-        return ids.entrySet().stream()
-                .map(entry -> Map.entry(entry.getKey(), getFilters(entry.getValue()).stream().map(f -> (long) getIdentifiableAttributes(f, networkUuid, variantId).size()).toList()))
+    public Map<String, Long> getIdentifiablesCountByGroup(IdsByGroup idsByGroup, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(idsByGroup);
+        return idsByGroup.getIds().entrySet().stream()
+                .map(entry -> Map.entry(entry.getKey(), getFilters(entry.getValue()).stream().map(f -> (long) getIdentifiableAttributes(f, networkUuid, variantId).size()).mapToLong(Long::longValue).sum()))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Comment (TheMaskedTurtle) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:703]:
```suggestion
        return idsByGroup.getIds().entrySet().stream()
                .collect(Collectors.toMap(
                    Map.Entry::getKey,
                    entry -> getFilters(entry.getValue()).stream()
                        .map(f -> getIdentifiableAttributes(f, networkUuid, variantId).size())
                        .mapToLong(Integer::longValue)
                        .sum()
                    )
                );
```

Comment (TheMaskedTurtle) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:703]: It could be collected toMap directly

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 2]
```diff
@@ -1012,31 +1007,28 @@ public void testGetIdentifiablesCount() throws Exception {
...
                         .andReturn().getResponse().getContentAsString(),
                 new TypeReference<>() {
                 });
-        assertEquals(1, filtersComplexityCount.get("0").size());
-        assertEquals(1, filtersComplexityCount.get("1").size());
-        assertEquals(1, filtersComplexityCount.get("2").size());
-        assertEquals(0, filtersComplexityCount.get("3").size());
+        assertEquals(1, identifiablesCount.get("g1").longValue());
+        assertEquals(1, identifiablesCount.get("g2").longValue());
+        assertEquals(0, identifiablesCount.get("g3").longValue());
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:1028]: Why filter of filterId3 is not returning 1 anymore ?

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 3]
```diff
@@ -830,8 +830,6 @@ public void testIdentifierListFilter() throws Exception {
 
         // Create identifier list filter for hvdc
         UUID hvdcFilterId = UUID.randomUUID();
-        IdentifierListFilterEquipmentAttributes hvdc1 = new IdentifierListFilterEquipmentAttributes("threeWT1", null);
-        IdentifierListFilterEquipmentAttributes hvdc2 = new IdentifierListFilterEquipmentAttributes("threeWT2", null);
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:834]: Oh I guess that's why filterId3 is empty now, why did you delete these ?

File: src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+@Data
+@Builder
+public class IdsByGroup {
```
Comment (TheMaskedTurtle) [inline src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java:21]: Maybe it's not necessary to define this type ?

File: src/main/java/org/gridsuite/filter/server/FilterController.java [hunk 5]
```diff
@@ -147,15 +147,15 @@ public ResponseEntity<List<IdentifiableAttributes>> exportFilter(@PathVariable("
...
-    @PostMapping(value = "/filters/identifiables-count", produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/filters/identifiables-count", produces = MediaType.APPLICATION_JSON_VALUE)
     @Operation(summary = "Calculate the total of identifiables for a list of filters")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Identifiables count")})
-    public ResponseEntity<Map<String, List<Long>>> getIdentifiablesCount(@RequestBody Map<String, List<UUID>> ids,
-                                                                            @RequestParam(value = "networkUuid") UUID networkUuid,
-                                                                            @RequestParam(value = "variantId", required = false) String variantId) {
+    public ResponseEntity<Map<String, Long>> getIdentifiablesCountByGroup(@RequestParam(value = "networkUuid") UUID networkUuid,
+                                                                          @RequestParam(value = "variantId", required = false) String variantId,
+                                                                          IdsByGroup idsByGroup) {
```
Comment (TheMaskedTurtle) [inline src/main/java/org/gridsuite/filter/server/FilterController.java:155]: You should probably add @RequestParam if you delete the class

**Review by TheMaskedTurtle — COMMENTED** [review]
In just an API point of view, I feel it's a bit weird to treat this endpoint as a map with input and output, why not just lists ? Actually just a list of UUID in input that returns a Long should even be enough for the unique use case of this endpoint

File: src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java (hunk 4 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java:21]: Map in a request param with Spring MVC has a special meaning.
See class documentation of `RequestParam`

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:1028]: It's not the same
`filtersComplexityCount` was a list (List.of(0))
`identifiablesCount` is the value in the list

**Comment by SlimaneAmar** [comment]
> In just an API point of view, I feel it's a bit weird to treat this endpoint as a map with input and output, why not just lists ? Actually just a list of UUID in input that returns a Long should even be enough for the unique use case of this endpoint

It's an optimization to reduce the number of requests
They are grouped

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/filter/server/FilterService.java:703]: yes

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:834]: Not used

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java (hunk 2 above)
Comment (TheMaskedTurtle) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:1028]: indeed sorry ! :)

File: src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java (hunk 4 above)
Comment (TheMaskedTurtle) [reply src/main/java/org/gridsuite/filter/server/dto/IdsByGroup.java:21]: Actually it works quite well as we expect for Map<string, string> or MultiValueMap<string, string> and @RequestParam, but yes with this it does not properly deserialize to UUIDs... so ok for the class.

**Comment by TheMaskedTurtle** [comment]
> > In just an API point of view, I feel it's a bit weird to treat this endpoint as a map with input and output, why not just lists ? Actually just a list of UUID in input that returns a Long should even be enough for the unique use case of this endpoint
> 
> It's an optimization to reduce the number of requests They are grouped

Then ok, in the sensitivity-analysis-server (which is the only caller from what I saw) we still ignore the keys of the Map, so we could still do one grouped request with List<List<UUID>> that returns a List<Long>, but if it's just that it doesn't change much so as you wish.

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 1 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/filter/server/FilterService.java]: done

**Review by TheMaskedTurtle — APPROVED** [review]
